src/signature/spki.rs

```diff
@@ -472,154 +449,139 @@ mod tests {
     test_parse_spki_bad_outer!(test_rsa_pkcs1_sha1_bad_key_der_length,
                                "rsa-pkcs1-sha1-bad-key-der-length.pem",
                                VerifyWithSPKIError::BadDER);
     test_parse_spki_bad_outer!(test_rsa_pkcs1_sha1_bad_key_der_null,
                                "rsa-pkcs1-sha1-bad-key-der-null.pem",
                                VerifyWithSPKIError::BadDER);
-    test_verify_signed_data!(test_rsa_pkcs1_sha1_key_params_absent,
+    test_verify_signature!(test_rsa_pkcs1_sha1_key_params_absent,
                              "rsa-pkcs1-sha1-key-params-absent.pem",
+                             &spki::RSA_PKCS1_2048_8192_SHA1,
                              Err(VerifyWithSPKIError::UnsupportedSignatureAlgorithmForPublicKey));
-    test_verify_signed_data!(
-        test_rsa_pkcs1_sha1_using_pss_key_no_params,
+    // We only support rsa keys identified as "rsaEncyrption", not rsa pss, so this is really only
+    // a test that "rsaEncryption" != "rsassapss", not about params.
+    test_verify_signature!( test_rsa_pkcs1_sha1_using_pss_key_no_params,
         "rsa-pkcs1-sha1-using-pss-key-no-params.pem",
+        &spki::RSA_PKCS1_2048_8192_SHA1,
         Err(VerifyWithSPKIError::UnsupportedSignatureAlgorithmForPublicKey));
-    test_verify_signed_data!(test_rsa_pkcs1_sha1_wrong_algorithm,
-                             "rsa-pkcs1-sha1-wrong-algorithm.pem",
-                             Err(VerifyWithSPKIError::InvalidSignatureForPublicKey));
     // XXX: RSA PKCS#1 with SHA-1 is a supported algorithm, but we only accept
     // 2048-8192 bit keys, and this test file is using a 1024 bit key. Thus,
     // our results differ from Chromium's. TODO: this means we need a 2048+ bit
     // version of this test.
-    test_verify_signed_data!(test_rsa_pkcs1_sha1, "rsa-pkcs1-sha1.pem",
-                             Err(VerifyWithSPKIError::InvalidSignatureForPublicKey));
+    test_verify_signature!(test_rsa_pkcs1_sha1,
+                           "rsa-pkcs1-sha1.pem",
+                           &spki::RSA_PKCS1_2048_8192_SHA1,
+                           Err(VerifyWithSPKIError::InvalidSignatureForPublicKey));
     // XXX: RSA PKCS#1 with SHA-1 is a supported algorithm, but we only accept
     // 2048-8192 bit keys, and this test file is using a 1024 bit key. Thus,
     // our results differ from Chromium's. TODO: this means we need a 2048+ bit
     // version of this test.
-    test_verify_signed_data!(test_rsa_pkcs1_sha256, "rsa-pkcs1-sha256.pem",
-                             Err(VerifyWithSPKIError::InvalidSignatureForPublicKey));
+    test_verify_signature!(test_rsa_pkcs1_sha256,
+                           "rsa-pkcs1-sha256.pem",
+                           &spki::RSA_PKCS1_2048_8192_SHA256,
+                           Err(VerifyWithSPKIError::InvalidSignatureForPublicKey));
+    // TODO this only tests DER decoding, not signature logic
     test_parse_spki_bad_outer!(test_rsa_pkcs1_sha256_key_encoded_ber,
```

Comment:
Works for me. I'll check to make sure that functionality has some tests but if so I'll delete these.